### PR TITLE
Runtime upgrade nodejs

### PR DESCRIPTION
**Pull request by alexkdefi**
## Issue
https://github.com/ComposableFi/composable/projects/2#card-80659106

## Description
GA action for pulling ansible playbooks. This one pull runtime upgrade

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: .maintain/subwasm.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import os
+import time
+import argparse
+import sys
+import json
+import requests
```
Comment (andor0) [inline .maintain/subwasm.py:6]: `requests` doesn't included in python batteries,  what is version needs? should add `requirements.txt`?

File: .maintain/subwasm.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+def main():
+
+    parser = argparse.ArgumentParser(description='wasm checker')
+    parser.add_argument('--chain',choices=['dali', 'composable', 'picasso'])
+    args = parser.parse_args(sys.argv[1:])
+
+    if(args.chain == 'dali'):
+        print('dali')
+        local_wasm = process_local_subwasm(chain='dali', path='/home/ubuntu/')
+        remote_wasm = process_remote_subwasm(http_con_str='http://localhost:9933')
```
Comment (andor0) [inline .maintain/subwasm.py:69]: imho, it can remove because you do this next

File: .maintain/subwasm.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            print(remote_wasm)
+            if remote_wasm == local_wasm:
+                SendMessageSlack(message='wasm updated!')
+                break
+
+
+    if(args.chain == 'composable'):
+        print('dali')
+        local_wasm = process_local_subwasm(chain='composable', path='/home/ubuntu/')
+        remote_wasm = process_remote_subwasm(http_con_str='http://localhost:9933')
```
Comment (andor0) [inline .maintain/subwasm.py:86]: imho, it can remove because you do this next

File: .maintain/subwasm.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            print(remote_wasm)
+            if remote_wasm == local_wasm:
+                SendMessageSlack(message='wasm updated!')
+                break
+
+
+    if(args.chain == 'picasso'):
+        print('dali')
+        local_wasm = process_local_subwasm(chain='picasso', path='/home/ubuntu/')
+        remote_wasm = process_remote_subwasm(http_con_str='http://localhost:9933')
```
Comment (andor0) [inline .maintain/subwasm.py:101]: imho, it can remove because you do this next

File: .maintain/subwasm.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            local_wasm = remote_wasm
+            print(local_wasm)
+            print(remote_wasm)
+            if remote_wasm == local_wasm:
+                SendMessageSlack(message='wasm updated!')
+                break
+
+
+    if(args.chain == 'composable'):
+        print('dali')
```
Comment (andor0) [inline .maintain/subwasm.py:84]:
```suggestion
        print('composable')
```

File: .maintain/subwasm.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            remote_wasm = process_remote_subwasm(http_con_str='http://localhost:9933')
+            print(local_wasm)
+            print(remote_wasm)
+            if remote_wasm == local_wasm:
+                SendMessageSlack(message='wasm updated!')
+                break
+
+
+    if(args.chain == 'picasso'):
+        print('dali')
```
Comment (andor0) [inline .maintain/subwasm.py:99]:
```suggestion
        print('picasso')
```

**Comment by andor0** [comment]
can do the `main` function simpler:
```
def update_wasm(chain):
        print(chain)
        local_wasm = process_local_subwasm(chain=chain, path='/home/ubuntu/')
        while True:
            time.sleep(15)
            remote_wasm = process_remote_subwasm(http_con_str='http://localhost:9933')
            local_wasm = remote_wasm
            print(local_wasm)
            print(remote_wasm)
            if remote_wasm == local_wasm:
                SendMessageSlack(message='wasm updated!')
                break

def main():
    parser = argparse.ArgumentParser(description='wasm checker')
    parser.add_argument('--chain',choices=['dali', 'composable', 'picasso'])
    args = parser.parse_args(sys.argv[1:])

    update_wasm(args.chain)
```

**Review by haroldsphinx — APPROVED** [review]

**Pull request closed without merging**